### PR TITLE
Automated Deployments include IMDS registration

### PR DESCRIPTION
**Pull request by tniek**
- Add trackers via Azure Metadata Service for beginning and completion of Ansible deployment
- Fix branch name to work with Azure Pipelines

### CODE REVIEW

File: deploy/vm/ansible/roles/imds-reg/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+# This role registers the current deployment state with Azure's Metadata Service (IMDS)
+- name: Register with IMDS
+  uri:
+    url: "http://169.254.169.254/metadata/instance?api-version={{ api_version }}"
+    method: GET
```
[the remark this answers is not in the document]
Comment (tniek) [reply deploy/vm/ansible/roles/imds-reg/tasks/main.yml]: Discussed details internally, GET is indeed the right method

**Comment by lagalbra** [comment]
Your PR description mentions "Fix branch name to work with Azure Pipelines"; does https://github.com/Azure/sap-hana/blob/master/CONTRIBUTING.md reflect how branches should be named?

**Comment by persiaAziz** [comment]
@lagalbra Yep, It is specified in the branch policy

**Comment by PakDLiu** [comment]
Question: Is there no other way to tag the operation? Is using User agent the best way for this? I thought user agent was used for what type of agent the user was using (IE: Windows, Ubuntu, etc)

**Comment by tniek** [comment]
> Your PR description mentions "Fix branch name to work with Azure Pipelines"; does https://github.com/Azure/sap-hana/blob/master/CONTRIBUTING.md reflect how branches should be named?

It does, I overlooked that statement at the very end:
> **So the branch name can only contain small case alphanumeric characters and hyphens.**

**Pull request merged**